typings/index.d.ts

```diff
@@ -779,33 +781,114 @@ export class GuildAuditLogs {
   public toJSON(): unknown;
 }
 
-export class GuildAuditLogsEntry {
+export class GuildAuditLogsEntry<
+  TAction extends GuildAuditLogsAction = 'ALL',
+  TActionType extends GuildAuditLogsActionType = TAction extends
+    | 'CHANNEL_CREATE'
+    | 'CHANNEL_OVERWRITE_CREATE'
+    | 'MEMBER_BAN_REMOVE'
+    | 'BOT_ADD'
+    | 'ROLE_CREATE'
+    | 'INVITE_CREATE'
+    | 'WEBHOOK_CREATE'
+    | 'EMOJI_CREATE'
+    | 'MESSAGE_PIN'
+    | 'INTEGRATION_CREATE'
+    | 'STAGE_INSTANCE_CREATE'
+    | 'STICKER_CREATE'
+    | 'THREAD_CREATE'
+    ? 'CREATE'
+    : TAction extends
+        | 'CHANNEL_DELETE'
+        | 'CHANNEL_OVERWRITE_DELETE'
+        | 'MEMBER_KICK'
+        | 'MEMBER_PRUNE'
+        | 'MEMBER_BAN_ADD'
+        | 'MEMBER_DISCONNECT'
+        | 'ROLE_DELETE'
+        | 'INVITE_DELETE'
+        | 'WEBHOOK_DELETE'
+        | 'EMOJI_DELETE'
+        | 'MESSAGE_DELETE'
+        | 'MESSAGE_BULK_DELETE'
+        | 'MESSAGE_UNPIN'
+        | 'INTEGRATION_DELETE'
+        | 'STAGE_INSTANCE_DELETE'
+        | 'STICKER_DELETE'
+        | 'THREAD_DELETE'
+    ? 'DELETE'
+    : TAction extends
+        | 'GUILD_UPDATE'
+        | 'CHANNEL_UPDATE'
+        | 'CHANNEL_OVERWRITE_UPDATE'
```

Comment:
These can use mapped types as well, although not as trivial, but can be accomplished like this:

```ts
export type AuditUpdateActions = {
  [key in keyof typeof AuditUpdateTypes]: 'UPDATE';
};

export type AuditDeleteActions = {
  [key in keyof typeof AuditDeleteTypes]: 'DELETE';
};

...

export type AllActions = AuditUpdateActions & AuditDeleteActions & ...;
```

Where `AuditDeleteTypes` and `AuditUpdateTypes` are enums with the subset of values that correspond to `'DELETE'` or `'UPDATE'`

Reply:
Great idea! I've done it in https://github.com/noftaly/discord.js/commit/65f08d25ed00717b458ce2fe15fea2ea17442cbd and expended it to TTargetType, even though I didn't exactly do it as you proposed to prevent too much duplication